src/openrct2-ui/windows/Research.cpp

```diff
@@ -443,70 +470,62 @@ static void WindowResearchFundingMousedown(WindowBase* w, WidgetIndex widgetInde
     Dropdown::SetChecked(currentResearchLevel, true);
 }
 
-/**
- *
- *  rct2: 0x0069DB6D
- */
-static void WindowResearchFundingDropdown(WindowBase* w, WidgetIndex widgetIndex, int32_t dropdownIndex)
+void WindowResearchFundingMouseUp(WidgetIndex widgetIndex, WidgetIndex baseWidgetIndex)
 {
-    if (widgetIndex != WIDX_RESEARCH_FUNDING_DROPDOWN_BUTTON || dropdownIndex == -1)
-        return;
+    // Offset the widget index to allow reuse from other windows
+    baseWidgetIndex = baseWidgetIndex - WIDX_RESEARCH_FUNDING;
 
-    auto gameAction = ParkSetResearchFundingAction(gResearchPriorities, dropdownIndex);
-    GameActions::Execute(&gameAction);
+    if (widgetIndex >= (WIDX_TRANSPORT_RIDES + baseWidgetIndex) && widgetIndex <= (WIDX_SCENERY_AND_THEMING + baseWidgetIndex))
```

Comment:
This seems more complicated than the `switch` will fallthrough

Reply:
I think this came from the Finance window's implementation. I agree, it is clearer as a `switch` statement, will change.